comms/EclMeta.h

```diff
@@ -0,0 +1,221 @@
+
+public:
+
+    BEGIN_CUNKNOWN
+    END_CUNKNOWN(CUnknown)
+
+    CEclFile(const boost::filesystem::wpath path)
+    {
+        m_path = path;
+        m_defs.clear();
+        m_imports.clear();
+    }
+    void UpdateDef(CEclDefinition *def)
+    {
+        m_defs[def->GetName()] = def;
+    }
+    void UpdateImport(CEclImport *import)
+    {
+        m_imports[import->GetName()] = import;
+    }
+    /*
+        Scenarios:
+        "MyMod" and presses ctrl+space
+        "MyModule." (should auto popup suggestions)
+        "MyModule.MyAttr" and presses ctrl+space
+        "MyModule.MyAttribute." (should auto popup suggestions)
+        Cursor in "MyModule.MyAttribute" and user presses F12
+        Mouse over tooltip "MyModule.MyAttribute"
+        
+        IMPORT $.MyModule;
+        IMPORT MyModule AS MyMod;
+        IMPORT $ AS MyMod;
+        (Where MyModule could be a folder an attr file or a library (like std))
+        */
+    bool GetDefStrings(const std::_tstring & definitionName, StdStringVector &set)
+    {
+        bool found = false;
+        for (std::map<std::wstring, StlLinked<CEclDefinition>>::iterator itr = m_defs.begin(); itr != m_defs.end(); ++itr)
+        {
+            if (boost::algorithm::equals(definitionName, itr->first)) {
+                for (std::map<std::wstring, std::wstring>::iterator fieldItr = itr->second->m_fields.begin(); fieldItr != itr->second->m_fields.end(); ++fieldItr)
+                {
+                    if (std::find(set.begin(), set.end(), fieldItr->first) == set.end())
+                    {
+                        set.push_back(fieldItr->first);
+                        found = true;
+                    }
+                }
+            }
+        }
+        return found;
+    }
+    CEclImport * GetImportRefs(const std::_tstring & importAsName)
+    {
+        for (std::map<std::wstring, StlLinked<CEclImport>>::iterator itr = m_imports.begin(); itr != m_imports.end(); ++itr)
+        {
+            if (boost::algorithm::iequals(importAsName, itr->first) && !boost::algorithm::equals(importAsName, itr->second->GetRef()))
+            {
+                return dynamic_cast<CEclImport*>(itr->second.get());
+            }
+        }
+        return NULL;
+    }
+    CEclDefinition * GetDefinitions(const std::_tstring & defName)
+    {
+        for (std::map<std::wstring, StlLinked<CEclDefinition>>::iterator itr = m_defs.begin(); itr != m_defs.end(); ++itr)
+        {
+            if (boost::algorithm::iequals(defName, itr->first))
+            {
+                return dynamic_cast<CEclDefinition*>(itr->second.get());
+            }
+        }
+        return NULL;
+    }
+};
+
+typedef std::map<std::wstring, StlLinked<CEclMetaData> > EclMetaDataMap;
+
+class CEclMeta
+{
+public:
+    EclMetaDataMap m_masterMeta;
+    std::vector<CString> m_autoTokens;
+    std::_tstring m_autoString;
+    std::_tstring m_autoLast;
+
+    void LoadMetaData(const WPathVector *folders);
```

Comment:
If possible pass a const ref

Reply:
Done

Reply:
@GordonSmith actually, this one has to stay a string because it uses "dots" instead of the normal path notation. This is what was breaking for me. I'm putting it back and retesting.